// luxon-angular/esm2015/lib/formatting/date-time-to-relative.pipe.js
import { __decorate } from "tslib";
import { ChangeDetectorRef, OnDestroy, Pipe, PipeTransform } from '@angular/core';
import { interval } from 'rxjs';
let DateTimeToRelativePipe = class DateTimeToRelativePipe {
    constructor(ref) {
        this.ref = ref;
    }
    ngOnDestroy() {
        if (this.subscription) {
            this.subscription.unsubscribe();
            this.subscription = null;
        }
    }
    transform(value, unit, style) {
        if (value == null) {
            if (this.subscription) {
                this.subscription.unsubscribe();
                this.subscription = null;
            }
            return null;
        }
        if (!this.subscription) {
            this.subscription = interval(1000).subscribe(() => this.ref.markForCheck());
        }
        return value.toRelative({
            style,
            unit
        });
    }
};
DateTimeToRelativePipe.ctorParameters = () => [
    { type: ChangeDetectorRef }
];
DateTimeToRelativePipe = __decorate([
    Pipe({
        name: 'dateTimeToRelative',
        pure: false
    })
], DateTimeToRelativePipe);
export { DateTimeToRelativePipe };
