// luxon-angular/esm2015/lib/formatting/date-time-to-relative-calendar.pipe.js
import { __decorate } from "tslib";
import { ChangeDetectorRef, OnDestroy, Pipe, PipeTransform } from '@angular/core';
import { interval } from 'rxjs';
let DateTimeToRelativeCalendarPipe = class DateTimeToRelativeCalendarPipe {
    constructor(ref) {
        this.ref = ref;
    }
    ngOnDestroy() {
        if (this.subscription) {
            this.subscription.unsubscribe();
            this.subscription = null;
        }
    }
    transform(value, unit) {
        if (value == null) {
            if (this.subscription) {
                this.subscription.unsubscribe();
                this.subscription = null;
            }
            return null;
        }
        if (!this.subscription) {
            this.subscription = interval(1000).subscribe(() => this.ref.markForCheck());
        }
        return value.toRelativeCalendar({
            unit
        });
    }
};
DateTimeToRelativeCalendarPipe.ctorParameters = () => [
    { type: ChangeDetectorRef }
];
DateTimeToRelativeCalendarPipe = __decorate([
    Pipe({
        name: 'dateTimeToRelativeCalendar',
        pure: false
    })
], DateTimeToRelativeCalendarPipe);
export { DateTimeToRelativeCalendarPipe };
