// luxon-angular/esm5/lib/formatting/date-time-to-relative.pipe.js
import { __decorate } from "tslib";
import { ChangeDetectorRef, OnDestroy, Pipe, PipeTransform } from '@angular/core';
import { interval } from 'rxjs';
var DateTimeToRelativePipe = /** @class */ (function () {
    function DateTimeToRelativePipe(ref) {
        this.ref = ref;
    }
    DateTimeToRelativePipe.prototype.ngOnDestroy = function () {
        if (this.subscription) {
            this.subscription.unsubscribe();
            this.subscription = null;
        }
    };
    DateTimeToRelativePipe.prototype.transform = function (value, unit, style) {
        var _this = this;
        if (value == null) {
            if (this.subscription) {
                this.subscription.unsubscribe();
                this.subscription = null;
            }
            return null;
        }
        if (!this.subscription) {
            this.subscription = interval(1000).subscribe(function () { return _this.ref.markForCheck(); });
        }
        return value.toRelative({
            style: style,
            unit: unit
        });
    };
    DateTimeToRelativePipe.ctorParameters = function () { return [
        { type: ChangeDetectorRef }
    ]; };
    DateTimeToRelativePipe = __decorate([
        Pipe({
            name: 'dateTimeToRelative',
            pure: false
        })
    ], DateTimeToRelativePipe);
    return DateTimeToRelativePipe;
}());
export { DateTimeToRelativePipe };
