// luxon-angular/esm5/lib/formatting/date-time-to-relative-calendar.pipe.js
import { __decorate } from "tslib";
import { ChangeDetectorRef, OnDestroy, Pipe, PipeTransform } from '@angular/core';
import { interval } from 'rxjs';
var DateTimeToRelativeCalendarPipe = /** @class */ (function () {
    function DateTimeToRelativeCalendarPipe(ref) {
        this.ref = ref;
    }
    DateTimeToRelativeCalendarPipe.prototype.ngOnDestroy = function () {
        if (this.subscription) {
            this.subscription.unsubscribe();
            this.subscription = null;
        }
    };
    DateTimeToRelativeCalendarPipe.prototype.transform = function (value, unit) {
        var _this = this;
        if (value == null) {
            if (this.subscription) {
                this.subscription.unsubscribe();
                this.subscription = null;
            }
            return null;
        }
        if (!this.subscription) {
            this.subscription = interval(1000).subscribe(function () { return _this.ref.markForCheck(); });
        }
        return value.toRelativeCalendar({
            unit: unit
        });
    };
    DateTimeToRelativeCalendarPipe.ctorParameters = function () { return [
        { type: ChangeDetectorRef }
    ]; };
    DateTimeToRelativeCalendarPipe = __decorate([
        Pipe({
            name: 'dateTimeToRelativeCalendar',
            pure: false
        })
    ], DateTimeToRelativeCalendarPipe);
    return DateTimeToRelativeCalendarPipe;
}());
export { DateTimeToRelativeCalendarPipe };
